Content.Server/Nuke/NukeCodePaperSystem.cs

```diff
@@ -25,55 +27,35 @@ private void OnMapInit(EntityUid uid, NukeCodePaperComponent component, MapInitE
             SetupPaper(uid);
         }
 
-        private void SetupPaper(EntityUid uid, EntityUid? station = null, PaperComponent? paper = null)
+        private void SetupPaper(EntityUid uid, EntityUid? station = null)
         {
-            if (!Resolve(uid, ref paper))
+            if (TryGetRelativeNukeCode(uid, out var paperContent, station))
             {
-                return;
-            }
-
-            var owningStation = station ?? _station.GetOwningStation(uid);
-            var transform = Transform(uid);
-
-            // Find the first nuke that matches the paper's location.
-            foreach (var nuke in EntityQuery<NukeComponent>())
-            {
-                if (owningStation == null && nuke.OriginMapGrid != (transform.MapID, transform.GridUid)
-                    || nuke.OriginStation != owningStation)
-                {
-                    continue;
-                }
-
-                paper.Content += $"{MetaData(nuke.Owner).EntityName} - {nuke.Code}";
-                break;
+                _paper.SetContent(uid, paperContent);
             }
         }
 
         /// <summary>
-        ///     Send a nuclear code to all communication consoles
+        ///     Send a nuclear code to all faxes
```

Comment:
all faxes -> all faxes on that station which are authorized to receive nuke codes.